### PR TITLE
yaml/yml label file if/else guards

### LINKED ISSUE #23715: custom label option appears twice
### Description

when adding a custom yaml label file, an option appears twice. the glob appears to be searching for files as well as file types 

### Gitea Version

1.19.0

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

```
#pwd
/var/lib/gitea/custom/options/label

#ls
test.yaml

#cat test.yaml
labels:
  - name: "foo/bar"  # name of the label that will appear in the dropdown
    exclusive: true # whether to use the exclusive namespace for scoped labels. scoped delimiter is /
    color: aabbcc # hex colour coding
    description: Some label # long description of label intent
  - name: "foo/baz"  # name of the label that will appear in the dropdown
    exclusive: true # whether to use the exclusive namespace for scoped labels. scoped delimiter is /
    color: aabbcc # hex colour coding
    description: Some other label # long description of label intent
```

### Git Version

2.40.0

### Operating System

linux (gentoo)

### How are you running Gitea?

rc-service 

### Database

None

### PR DESCRIPTION
**Pull request by eeyrjmr**
ensure the legacy label parser is not called if a yaml file is detected.  fixes #23715 

Signed-off-by: Jon Roadley-Battin [jon.roadleybattin@gmail.com](mailto:jon.roadleybattin@gmail.com)

### CODE REVIEW

File: modules/label/parser.go [hunk 1]
```diff
@@ -36,22 +37,25 @@ func (err ErrTemplateLoad) Error() string {
...
-
-	data, err = options.Labels(name + ".yml")
-	if err == nil && len(data) > 0 {
-		return parseYamlFormat(name+".yml", data)
-	}
-
-	data, err = options.Labels(name)
-	if err != nil {
-		return nil, ErrTemplateLoad{name, fmt.Errorf("GetRepoInitFile: %w", err)}
+	if filepath.Ext(name) == ". yaml" {
```
Comment (wxiaoguang) [inline modules/label/parser.go]: Some questions:

1. `". yaml"`:  a space in it?
2. Gitea has builtin `Advanced.yaml`, if a user puts a customized `Advance.yml`, then still the builtin `Advanced.yaml` is used?

Comment (eeyrjmr) [reply modules/label/parser.go]: 1. Typo (will correct )
2.  Good question. I need to check but the built-in Advanced.yaml isn't in the custom options directory. It does raise the question that if some adds foo.yaml and foo.yml ... What should be used

File: modules/label/parser.go [hunk 2]
```diff
@@ -36,22 +37,25 @@ func (err ErrTemplateLoad) Error() string {
...
+		data, err = options.Labels(name + ".yml")
+		if err == nil && len(data) > 0 {
+			return parseYamlFormat(name+".yml", data)
+		}
+	} else {
+		data, err = options.Labels(name)
+		if err != nil {
+			return nil, ErrTemplateLoad{name, fmt.Errorf("GetRepoInitFile: %w", err)}
+		}	
+		return parseLegacyFormat(name, data)
```
Comment (zeripath) [inline modules/label/parser.go:55]: No need for the else

```suggestion
	}

	data, err = options.Labels(name)
	if err != nil {
		return nil, ErrTemplateLoad{name, fmt.Errorf("GetRepoInitFile: %w", err)}
	}	
	return parseLegacyFormat(name, data)
```

**Comment by zeripath** [comment]
I don't think this is right.

Is the intention here that if `name` has a suffix `.yaml` then `parseYamlFormat` will be run on that file? Because that is not what this is doing...

**Comment by eeyrjmr** [comment]
> I don't think this is right.
> 
> Is the intention here that if `name` has a suffix `.yaml` then `parseYamlFormat` will be run on that file? Because that is not what this is doing...
> 
> See #23719

The intent was if the suffix was yaml then execute the parser but I don't think your proposed solution actually solves the problem. 
It might be best to close this MR and continue the discussion in your's

**Pull request closed without merging**